tests/system_tests/conftest.py

```diff
@@ -861,10 +862,61 @@ def debug(wandb_debug, fixture_fn, base_url):
         yield None
 
 
+@pytest.fixture(scope="session")
+def wandb_backend_proxy_server(
+    base_url,
+) -> Generator[WandbBackendProxy, None, None]:
+    """Session fixture that starts up a proxy server for the W&B backend."""
+    base_url_parsed = urllib.parse.urlparse(base_url)
+
+    with spy_proxy(
+        proxy_port=8000,
+        target_host=base_url_parsed.hostname,
+        target_port=base_url_parsed.port,
+    ) as proxy:
+        yield proxy
+
+
+@pytest.fixture(scope="function")
+def wandb_backend_spy(
+    user,
+    wandb_backend_proxy_server: WandbBackendProxy,
+    monkeypatch: pytest.MonkeyPatch,
+) -> Generator[WandbBackendSpy, None, None]:
+    """Fixture that allows spying on requests to the W&B backend.
+
+    This patches WANDB_BASE_URL and creates a fake user for the test
+    setting auth-related environment variables.
+
+    NOTE: This replaces the `relay_server` fixture.
+
+    Usage:
+
+        def test_something(wandb_backend_spy):
+            with wandb.init() as run:
+                run.log({"x": 1})
+
+            with wandb_backend_spy.freeze() as snapshot:
+                history = snapshot.history(run_id=run.id)
+                assert history[0]["x"] == 1
+    """
+
+    # Use a fake API key for the test.
+    _ = user
+
+    # Connect to the proxy to spy on requests:
+    monkeypatch.setenv("WANDB_BASE_URL", "http://127.0.0.1:8000")
```

Comment:
should the port defined as "global" const? as it is used in 2 places here

Reply:
done